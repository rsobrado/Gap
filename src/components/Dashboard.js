import React, { useState, useEffect } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

import FullWidthTabs from './FullWidthTabs'
import PrimaryAppBar from './PrimaryAppBar'
import data from './../data/mattresses.json'


const useStyles = makeStyles((theme) => ({
  mattressesContaner: {
    padding: 20,
    '@media screen and (min-width: 600px)': {
      padding: 50,
    },
  },
  contained: {
    marginTop: 40,
    backgroundColor: 'rgb(230,180,22)',
    borderRadius: 10,
    color: '#282c34',
    textShadow: 'none',
    lineHeight: '1.5 !important',
    fontSize: '1.05rem !important',
    '&:hover': {
      backgroundColor: 'rgb(255,190,0)',
      color: '#282c34 !important',
    },
  },
}))

export default function Dashboard() {
  const classes = useStyles()
  const [mattresses, setMattresses] = useState(data.mattresses)
  const [cartItems, setCartItems] = useState(0)
  const [currentMattress, setCurrentMattress] = useState('classic')

  useEffect(() => {
    setMattresses(data.mattresses)
    setCartItems(0)
  }, [])
  const handleAddToCart = (event) => {
    setCartItems(cartItems+1)
  }
  const handleMattressChange = (value) => {
    if(value===0){
      setCurrentMattress('classic')
    }
    if(value===1){
      setCurrentMattress('loom')
    }
    if(value===2){
      setCurrentMattress('zen')
    }
  }


  return (
    <React.Fragment>
      <PrimaryAppBar cartItems={cartItems} />
      <Container fixed maxWidth="xl" className={classes.mattressesContaner}>
        <Grid container spacing={8}>
          <Grid item xs={12} lg={7}>
            <img
              src={'./images/' + currentMattress + '.jpg'}
              alt="mattress"></img>
          </Grid>
          <Grid item xs={12} lg={5} style={{ textAlign: 'left' }}>
            <Typography variant={'h3'}>Choose Your Mattress</Typography>
            {/* <Typography variant={'h6'}>Select Mattress Type</Typography> */}
            <FullWidthTabs
              mattresses={mattresses}
              onClick={handleMattressChange}
            />
            <Grid item xs={12} lg={12} style={{ textAlign: 'right' }}>
              <Button
                variant="contained"
                color="primary"
                onClick={handleAddToCart}
                classes={{ contained: classes.contained }}>
                Add to Cart
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Container>
    </React.Fragment>
  )
}
