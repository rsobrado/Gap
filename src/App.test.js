import React from 'react'
import { render } from '@testing-library/react'
import Button from '@material-ui/core/Button'

import { shallow } from 'enzyme'

import App from './App'
import Dashboard from './components/Dashboard'

test('Call to action Add to Cart exists', () => {
  const { getByText } = render(<Dashboard />)
  const linkElement = getByText(/Add to Cart/i)
  expect(linkElement).toBeInTheDocument()
})

describe('Dashboard', () => {
  it('should be defined', () => {
    expect(Dashboard).toBeDefined()
  })
  it('should contains Add to Cart CTA', () => {
    const mockCallBack = jest.fn()
    const button = shallow(<Button id="add-to-cart" onClick={mockCallBack} />)
    expect(button).toMatchSnapshot()
  })
  it('test Add to Cart CTA', () => {
    const mockCallBack = jest.fn()
    const button = shallow(<Button id="add-to-cart" onClick={mockCallBack} />)

    button.simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })
  
})
