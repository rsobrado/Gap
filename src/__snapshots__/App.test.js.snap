// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should con correctly 1`] = `ShallowWrapper {}`;

exports[`Dashboard should contains  correctly 1`] = `ShallowWrapper {}`;

exports[`Dashboard should contains Add to Cart CTA 1`] = `ShallowWrapper {}`;

exports[`Dashboard should contains Add to Cart CTA 2`] = `ShallowWrapper {}`;

exports[`Dashboard should contains Add to Cart correctly 1`] = `ShallowWrapper {}`;

exports[`Dashboard should render correctly 1`] = `ShallowWrapper {}`;
